contracts/utils/AssociatedArrayLib.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+library AssociatedArrayLib {
```

Comment:
refactored this and added new features.

I removed the "contains" function. for this its better to use the EnumerableSet4337